ipa-core/src/ff/prime_field.rs

```diff
@@ -518,6 +537,14 @@ mod fp32bit {
 mod fp61bit {
     field_impl! { Fp61BitPrime, u64, u128, 61, 2_305_843_009_213_693_951 }
 
+    // For multiply-accumulate of `Fp61BitPrime` using `u128` as the accumulator, we can add 64
+    // products onto an original field element before reduction is necessary. i.e., 64 * (2^61 - 2)^2 +
+    // (2^61 - 2) < 2^128.
```

Comment:
is the math here correct? I was wrong many times last couple of weeks, it feels that a sum of 64 products would be 64 * (2^61 - 1)^2

Reply:
2^61 - 1 is the prime, so 2^61 - 2 is the largest value a field element can have.

That aside, the math here is slightly different than before, because this is implementing an accumulator. The maximum value adds an additional 2^61 - 2 (not squared), representing the initial value of the accumulator (or the value after the previous reduction).

I did add some tests related to this, but that's not necessarily going to catch problems, since there are assumptions about the conditions where the maximum value occurs baked into the test.